docs/network-backends.md

```diff
@@ -1,3 +1,268 @@
 # Network Backends
 
-TODO
+The API layer at which `httpcore` interacts with the network is described as the network backend. Various backend implementations are provided, allowing `httpcore` to handle networking in different runtime contexts.
+
+## Working with network backends
+
+### The default network backend
+
+Typically you won't need to specify a network backend, as a default will automatically be selected. However, understanding how the network backends fit in may be useful if you want to better understand the underlying architecture. Let's start by seeing how we can explicitly select the network backend.
+
+First we're making a standard HTTP request, using a connection pool:
+
+```python
+import httpcore
+
+with httpcore.ConnectionPool() as http:
+    response = http.request('GET', 'https://www.example.com')
+```
+
+We can also have the same behavior, but be explicit with our selection of the network backend:
+
+```python
+import httpcore
+
+network_backend = httpcore.NetworkBackend()
+with httpcore.ConnectionPool(network_backend=network_backend) as http:
+    response = http.request('GET', 'https://www.example.com')
+```
+
+The `httpcore.NetworkBackend()` implementation handles the opening of TCP connections, and operations on the socket stream, such as reading, writing, and closing the connection.
+
+We can get a better understanding of this by using a network backend to send a basic HTTP/1.1 request directly:
+
+```python
+import httpcore
+import ssl
+import certifi
+
+# Create an SSL context using 'certifi' for the certificates.
+ssl_context = ssl.create_default_context()
+ssl_context.load_verify_locations(certifi.where())
+
+# A basic HTTP/1.1 request as a plain bytestring.
+request = b''.join([
+    b'GET / HTTP/1.1\r\n',
+    b'Host: www.example.com\r\n',
+    b'Accept: */*\r\n',
+    b'Connection: close\r\n',
+    b'\r\n'
+])
```

Comment:
Should we use a tuple here?

```suggestion
request = (
    b'GET / HTTP/1.1\r\n'
    b'Host: www.example.com\r\n'
    b'Accept: */*\r\n'
    b'Connection: close\r\n'
    b'\r\n'
)
```

Reply:
No, the \r\n is supposed to be in the beginning of the join

Reply:
> No, the \r\n is supposed to be in the beginning of the join

True, that'd be neater. I've done that now.